web/courses.go

```diff
@@ -38,11 +38,15 @@ func (s *AutograderService) createEnrollment(request *pb.Enrollment) error {
 }
 
 // updateEnrollment accepts or rejects a user to enroll in a course.
-func (s *AutograderService) updateEnrollment(ctx context.Context, sc scm.SCM, request *pb.Enrollment) error {
+func (s *AutograderService) updateEnrollment(ctx context.Context, sc scm.SCM, curUser string, request *pb.Enrollment) error {
 	enrollment, err := s.db.GetEnrollmentByCourseAndUser(request.CourseID, request.UserID)
 	if err != nil {
 		return err
 	}
+	// log changes to teacher status
+	if enrollment.Status == pb.Enrollment_TEACHER || request.Status == pb.Enrollment_TEACHER {
+		s.logger.Debugf("User %s attempts changing enrollment status of user %d from %s to %s", curUser, enrollment.UserID, enrollment.Status, request.Status)
```

Comment:
I'm still thinking about a way to avoid passing `curUser` just to print a debug statement...

Reply:
Yeah, I'm not very fond of it either. It is just that we have information about the current user at `autograder_service` level, but no information about the database user/enrollment entry, and vice versa.